data-prepper-api/src/main/java/org/opensearch/dataprepper/model/event/DefaultEventHandle.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.model.event;
+
+import org.opensearch.dataprepper.model.acknowledgements.AcknowledgementSet;
+import java.lang.ref.WeakReference;
+
+import java.time.Instant;
+import java.io.Serializable;
+
+public class DefaultEventHandle implements EventHandle, Serializable {
+    private Instant externalOriginationTime;
+    private final Instant internalOriginationTime;
+    private WeakReference<AcknowledgementSet> acknowledgementSetRef;
+
+    public DefaultEventHandle(final Instant internalOriginationTime) {
+        this.acknowledgementSetRef = null;
+        this.externalOriginationTime = null;
+        this.internalOriginationTime = internalOriginationTime;
+    }
+
+    @Override
+    public void setAcknowledgementSet(final AcknowledgementSet acknowledgementSet) {
+        this.acknowledgementSetRef = new WeakReference<>(acknowledgementSet);
+    }
+
+    @Override
+    public void setExternalOriginationTime(final Instant externalOriginationTime) {
+        this.externalOriginationTime = externalOriginationTime;
+    }
+
+    public AcknowledgementSet getAcknowledgementSet() {
+        if (acknowledgementSetRef == null) {
+            return null;
+        }
+        return acknowledgementSetRef.get();
+    }
+
+    @Override
+    public Instant getInternalOriginationTime() {
+        return this.internalOriginationTime;
+    }
+
+    @Override
```

Comment:
For the next PR let's scope down and first only implement internal end to end latency metric. For ddb use case we could potentially use this value for external (https://docs.aws.amazon.com/amazondynamodb/latest/APIReference/API_streams_StreamRecord.html#DDB-Type-streams_StreamRecord-ApproximateCreationDateTime), so we may not need to make any hasty decisions on users configuring this time. @dlvenable What do you think?

Reply:
If I understand your proposal, the `Source` will set this value on the Event. We don't need to let the user configure this in the pipeline. But, we could in the future. Right?

Reply:
OK. Will only implement internal latency for now in the OpenSearchSink and S3 Sink